smarts/core/smarts.py

```diff
@@ -389,6 +391,28 @@ def add_agent_with_mission(
                 f"Unable to add entry trap for new agent '{agent_id}' with mission."
             )
 
+    def trap_history_vehicles(
+        self,
+        vehicles_to_trap: Dict[str, Tuple[str, AgentSpec]],
+        veh_missions: Dict[str, Tuple[Mission, Mission]],
+    ):
+        for veh_id, (agent_id, agent_spec) in vehicles_to_trap.items():
+            # Create trap to be triggered immediately
+            try:
+                self.add_agent_with_mission(
+                    agent_id, agent_spec.interface, veh_missions[veh_id][0]
+                )
+            except PlanningError:
+                self._log.warning(
+                    f"Unable to create PositionalGoal for vehicle {veh_id}, falling back to TraverseGoal"
+                )
```

Comment:
This should be very rare, right?  I mean, given that the real vehicle from the history data was able to go from the mission start to the mission end, we should be able to find a route to get there too.   I guess this would indicate that we have an incomplete reconstruction of the map.

Did you guys actually encounter this exception in your testing?   Are there other reasons (aside from an incomplete map) that it might be thrown?

Reply:
Yes, we encountered it only for one of the vehicle we hijacked, though the reason for that was the incomplete map and detecting wrong end lane due it being hijacked at middle of the intersection. Since its mission was very short, we decided to fall back to Traversal Goal for it,